test/Test/Db.hs

```diff
@@ -1,2 +1,123 @@
-module Test.Db where
-    
\ No newline at end of file
+module Test.Db 
+       ( dbSpecs
+       ) where
+
+import IW.App (AppEnv, App)
+import IW.Core.Issue (Issue)
+import IW.Core.Repo (Repo)
+import IW.Db (getIssues, getRepos, upsertIssues, upsertRepos)
+
+import Test.Assert (equals, succeeds)
+import Test.Common (joinSpecs)
+import Test.Data (invalidIssue, updatedValidIssue, updatedValidRepo, validIssue, validRepo)   
+import Test.Hspec (Spec, describe, it)
+
+
+dbSpecs :: AppEnv -> Spec
+dbSpecs = joinSpecs "Databse SQL query correctness"
+    [ repoSpecs
+    , issueSpecs
+    ]
+
+-- | Takes a getter function for getting values from the environment before and after an action, 
+-- a function for comparing those values, and the action
+beforeAfter :: App a -> (a -> a -> b) -> App c -> App b
+beforeAfter getter comparison action = do
+    before <- getter 
+    _  <- action
+    after <- getter
+    pure $ after `comparison` before
+
+---------------
+-- REPO SPEC --
+---------------
+
+repoSpecs :: AppEnv -> Spec
+repoSpecs = joinSpecs "Repo"
+    [ upsertReposSpec
+    , getReposSpec 
+    ]
+
+upsertReposSpec :: AppEnv -> Spec
+upsertReposSpec env = describe "upsertRepos" $ do
+    it "should do nothing when inserting an empty list" $ do
+        env & succeeds (upsertRepos [])
+        env & reposAddedRows (upsertRepos []) `equals` 0
+    it "should insert repos if the repos are valid" $
+        env & reposIncreased (upsertRepos [validRepo]) `equals` True
+    it "should update the repo if the same repo already exists" $
+        env & reposRowDifference (upsertRepos [updatedValidRepo]) `equals` [updatedValidRepo]
```

Comment:
@rashadg1030 You've implemented pretty nice mini-testing framework inside `issue-wanted` :+1: Great job :clap:

Reply:
Thank you😄